utils/build-script

```diff
@@ -527,9 +527,11 @@ class BuildScriptInvocation(object):
             # just invoke cmake from build-script directly rather than futzing
             # with build-script-impl. This makes even more sense since there
             # really isn't a security issue here.
-            impl_args += [
-                "--%s-cmake-options=%s" % (product_name, ' '.join(cmake_opts))
-            ]
+            if cmake_opts:
+                impl_args += [
+                    "--%s-cmake-options=%s" %
```

Comment:
We should be using the more modern Python string formatting via the `str.format` method. This could be re-written as: `'--{}-cmake-options={}'.format(product_name, ' '.join(cmake_opts))`.